plugins/catalog-backend/config.d.ts

```diff
@@ -38,6 +38,33 @@ export interface Config {
        * E.g. ["Component", "API", "Template", "Location"]
        */
       allow: Array<string>;
+      /**
+       * Limit this rule to a specific location
+       *
+       * Example with a fixed location
+       *  { "type": "url", "exact": "https://github.com/a/b/blob/file.yaml"}
+       *
+       * Example using a Regex
+       *  { "type": "url", "pattern": "https://github.com/org/*\/blob/master/*.yaml"}
+       *
+       * Using both exact and pattern will result in an error starting the application
+       */
+      locations?: Array<{
+        /**
+         * The type of location, e.g. "url".
+         */
+        type: string;
```

Comment:
Could this be optional?

Reply:
Yes! I'll just have to make some slight changes

Reply:
Tbh I think it's best to keep it required, just to make sure it's always explicit. I'm thinking for example if you have url-like custom locations and you accidentally allow URLs from an unintended source too.

Reply:
Okay. I can change it back